test/e2e/e2ethanos/services.go

```diff
@@ -259,30 +247,63 @@ func NewReceiverWithConfigWatcher(sharedDir string, networkName string, name str
 		DefaultImage(),
 		// TODO(bwplotka): BuildArgs should be interface.
 		e2e.NewCommand("receive", e2e.BuildArgs(map[string]string{
-			"--debug.name":                              fmt.Sprintf("receive-%v", name),
-			"--grpc-address":                            ":9091",
-			"--grpc-grace-period":                       "0s",
-			"--http-address":                            ":8080",
-			"--remote-write.address":                    ":8081",
-			"--label":                                   fmt.Sprintf(`receive="%s"`, name),
-			"--tsdb.path":                               filepath.Join(container, "data"),
-			"--log.level":                               infoLogLevel,
-			"--receive.replication-factor":              strconv.Itoa(replicationFactor),
-			"--receive.local-endpoint":                  localEndpoint,
-			"--receive.hashrings-file":                  filepath.Join(container, "hashrings.json"),
-			"--receive.hashrings-file-refresh-interval": "5s",
+			"--debug.name":        fmt.Sprintf("receive-%v", name),
+			"--grpc-address":      ":9091",
+			"--grpc-grace-period": "0s",
+			"--http-address":      ":8080",
+			"--label":             fmt.Sprintf(`receive="%s"`, name),
+			"--tsdb.path":         filepath.Join(container, "data"),
+			"--log.level":         infoLogLevel,
 		})...),
 		e2e.NewHTTPReadinessProbe(8080, "/-/ready", 200, 200),
 		8080,
 		9091,
-		8081,
 	)
 	receiver.SetUser(strconv.Itoa(os.Getuid()))
 	receiver.SetBackoff(defaultBackoffConfig)
-
 	return receiver, nil
 }
 
+func NewReceiveRouter(sharedDir string, networkName string, name string, replicationFactor int, hashring ...route.HashringConfig) (*e2e.HTTPService, error) {
+
+	dir := filepath.Join(sharedDir, "data", "receive-route", name)
+	container := filepath.Join(e2e.ContainerSharedDir, "data", "receive-route", name)
+	if err := os.MkdirAll(dir, 0777); err != nil {
+		return nil, errors.Wrap(err, "create router dir")
+	}
+	b, err := json.Marshal(hashring)
+	if err != nil {
+		return nil, errors.Wrapf(err, "generate hashring file: %v", hashring)
+	}
+
+	if err := ioutil.WriteFile(filepath.Join(dir, "hashrings.json"), b, 0666); err != nil {
+		return nil, errors.Wrap(err, "creating router config")
+	}
+
+	router := e2e.NewHTTPService(
+		fmt.Sprintf("receive-route-%v", name),
+		DefaultImage(),
+		// TODO(bwplotka): BuildArgs should be interface.
+		e2e.NewCommand("receive-route", e2e.BuildArgs(map[string]string{
+			"--debug.name":                 fmt.Sprintf("receive-route-%v", name),
+			"--http-address":               ":8080",
+			"--remote-write.address":       ":8081",
+			"--label":                      fmt.Sprintf(`receive="%s"`, name),
+			"--tsdb.path":                  filepath.Join(container, "data"),
```

Comment:
Looks like the receive router doesn't have `--label`  and `--tsdb.path` flags

Reply:
Thanks for pointing out!